Added localized box detection algorithm for digits.

anissingh opened the pull request:
## What?

Added support for box detection when only a localized part of the student information section of the cover page is selected in the MarkUs interface. These changes currently only work with digit detection for the student number section of the cover page.

## Why?

The original algorithm would crash often when it was not provided with the full student information section. However, the MarkUs interface seems to expect only a small, more localized section to be uploaded. These changes ensure consistency between the web interface and the expectations of the image processing scripts in this repository.

## How?

The `cv2` library in Python was used to detect square contours in the image. This involves using bounding boxes of contours and attempting to approximate them as squares. Erroneous contours (non-box contours) are then removed so as to leave only the box contours.

david-yz-liu reviewed (commented):
@anissingh great work. My only comment is that you should revert the changmes made in `cnn.py`, as they are not related to your work. You probably committed them accidentally.

anissingh commented:
@david-yz-liu thanks for catching that, I've reverted the changes to `cnn.py`. I've also added some morphological operations to create an image containing only vertical and horizontal lines of the original image, and then use this image to detect box contours. These changes seem to improve box detection. In hindsight, I probably should have waited to complete this pull request before incorporating these changes. I will make sure I don't try to do too many things at once in the future.
